src/ripple/nodestore/DatabaseShard.h

```diff
@@ -0,0 +1,135 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_NODESTORE_DATABASESHARD_H_INCLUDED
+#define RIPPLE_NODESTORE_DATABASESHARD_H_INCLUDED
+
+#include <ripple/nodestore/Database.h>
+#include <ripple/app/ledger/Ledger.h>
+#include <ripple/nodestore/Types.h>
+
+#include <boost/optional.hpp>
+
+#include <memory>
+
+namespace ripple {
+namespace NodeStore {
+
+/** Finds a shard index containing the ledger sequence
+
+    @param seq The ledger sequence
+    @return The shard index
+*/
+constexpr
+std::uint32_t
+seqToShardIndex(std::uint32_t const seq)
+{
+    return (seq - 1) / ledgersPerShard;
```

Comment:
Not a sequence number we should see, but if `seq=0` this will not return the expected result.

Reply:
True, I'll add an assert.